docs/concepts/projects/dependencies.md

```diff
@@ -226,6 +226,11 @@ The following dependency sources are supported by uv:
     project tables will be used. If another tool is being used for development, any metadata
     provided in the source table will need to be re-specified in the other tool's format.
 
+!!! important
+
+    Like other development-only features, sources defined by your dependencies won't ever be
+    enabled, even when building your project in development mode.
```

Comment:
I am hesitant about two important blocks in a row. I find a couple things confusing, like we haven't really discussed "development-only features" and I don't know what "building your project in development mode" means.

I'd probably just add a new paragraph above like...

> Since sources are specific to uv, they are also not propagated to package distributions, such as when using `uv build`.

Reply:
That seems a bit opaque for the average user.